dev/NavigationView/NavigationViewItemPresenter.cpp

```diff
@@ -7,18 +7,28 @@
 #include "NavigationViewItem.h"
 #include "SharedHelpers.h"
 
+static constexpr auto c_iconBoxColumnDefinitionName = L"IconColumn"sv;
+
 NavigationViewItemPresenter::NavigationViewItemPresenter()
 {
     SetDefaultStyleKey(this);
 }
 
 void NavigationViewItemPresenter::OnApplyTemplate()
 {
+    winrt::IControlProtected controlProtected = *this;
+
     // Retrieve pointers to stable controls 
     m_helper.Init(*this);
     if (auto navigationViewItem = GetNavigationViewItem())
     {
         navigationViewItem->UpdateVisualStateNoTransition();
+
+        // We probably switched displaymode, so restore width now!
+        if (navigationViewItem->GetNavigationView().PaneDisplayMode() != winrt::NavigationViewPaneDisplayMode::Top)
+        {
+            UpdateCompactPaneLength(m_compactPaneLengthValue, true);
+        }
```

Comment:
probably?

Reply:
When switching displaymode, we also reapply template. If we don't update the width here, it only will happen the next time the CompactPaneLength changes.